File: backend/src/main/java/backend/model/dto/request/RegisterRequest.java
package backend.model.dto.request;

import jakarta.validation.constraints.*;

public record RegisterRequest(
        @NotBlank(message = "fullname is required")
        String fullname,
        String identityNumber,
        Boolean confirmLinkRequest,
        @NotBlank(message = "Email is required")
        @Email(message = "Email format is invalid")
        String email,
        @NotBlank(message = "Phone is required")
        String phone,
        @NotBlank(message = "Password is required")
        @Size(min = 6, max = 20, message = "Password must be between 8 and 20 characters")
        String password
) {}

File: frontend/lib/data/models/auth/register_request.dart
import 'dart:convert';

class RegisterRequest {
  final String fullname;
  final String identityNumber;
  final bool? confirmLinkRequest;
  final String email;
  final String phone;
  final String password;

  RegisterRequest({
    required this.fullname,
    required this.identityNumber,
    this.confirmLinkRequest,
    required this.email,
    required this.phone,
    required this.password,
  });

  Map<String, dynamic> toMap() {
    final Map<String, dynamic> map = {
      'fullname': fullname,
      'identityNumber': identityNumber,
      'email': email,
      'phone': phone,
      'password': password,
    };
    if (confirmLinkRequest != null) {
      map['confirmLinkRequest'] = confirmLinkRequest;
    }
    return map;
  }

  String toJson() => JsonEncoder().convert(toMap());

  factory RegisterRequest.fromMap(Map<String, dynamic> map) {
    return RegisterRequest(
      fullname: map['fullname'] ?? '',
      identityNumber: map['identityNumber']?.toString() ?? '',
      confirmLinkRequest: map['confirmLinkRequest'] as bool?,
      email: map['email'] ?? '',
      phone: map['phone'] ?? '',
      password: map['password'] ?? '',
    );
  }
}
